orm/src/test/java/org/hibernate/search/test/query/facet/MultiValuedFacetingTest.java

```diff
@@ -0,0 +1,354 @@
+		NumberCollectionFacetEntity entity = new NumberCollectionFacetEntity( 1L );
+		storeData( entity.id, entity );
+		entity = new NumberCollectionFacetEntity( 1L, 42.2f );
+		storeData( entity.id, entity );
+		entity = new NumberCollectionFacetEntity( 2L, 42.3f, 442.2f );
+		storeData( entity.id, entity );
+
+		HSQuery hsQuery = sfHolder.getSearchFactory().createHSQuery( new MatchAllDocsQuery(), NumberCollectionFacetEntity.class );
+
+		QueryBuilder builder = sfHolder.getSearchFactory().buildQueryBuilder().forEntity( NumberCollectionFacetEntity.class ).get();
+		FacetingRequest facetReq = builder.facet()
+				.name( "someFacet" )
+				.onField( "facet" )
+				.range()
+				.from( 0.0f ).to( 100.0f ).excludeLimit()
+				.from( 100.0f ).to( 500.0f ).excludeLimit()
+				.orderedBy( FacetSortOrder.COUNT_DESC )
+				.includeZeroCounts( false )
+				.maxFacetCount( 10 )
+				.createFacetingRequest();
+
+		List<Facet> facets = hsQuery.getFacetManager().enableFaceting( facetReq ).getFacets( "someFacet" );
+		assertEquals( "There should be two facets", 2, facets.size() );
+		assertFacet( facets.get( 0 ), "[0.0, 100.0)", 2 );
+		assertFacet( facets.get( 1 ), "[100.0, 500.0)", 1 );
+	}
+
+	@Test
+	public void numberMap() throws Exception {
+		NumberMapFacetEntity entity = new NumberMapFacetEntity( 1L );
+		storeData( entity.id, entity );
+		entity = new NumberMapFacetEntity( 1L, 42.2f );
+		storeData( entity.id, entity );
+		entity = new NumberMapFacetEntity( 2L, 42.3f, 442.2f );
+		storeData( entity.id, entity );
+
+		HSQuery hsQuery = sfHolder.getSearchFactory().createHSQuery( new MatchAllDocsQuery(), NumberMapFacetEntity.class );
+
+		QueryBuilder builder = sfHolder.getSearchFactory().buildQueryBuilder().forEntity( NumberMapFacetEntity.class ).get();
+		FacetingRequest facetReq = builder.facet()
+				.name( "someFacet" )
+				.onField( "facet" )
+				.range()
+				.from( 0.0f ).to( 100.0f ).excludeLimit()
+				.from( 100.0f ).to( 500.0f ).excludeLimit()
+				.orderedBy( FacetSortOrder.COUNT_DESC )
+				.includeZeroCounts( false )
+				.maxFacetCount( 10 )
+				.createFacetingRequest();
+
+		List<Facet> facets = hsQuery.getFacetManager().enableFaceting( facetReq ).getFacets( "someFacet" );
+		assertEquals( "There should be two facets", 2, facets.size() );
+		assertFacet( facets.get( 0 ), "[0.0, 100.0)", 2 );
+		assertFacet( facets.get( 1 ), "[100.0, 500.0)", 1 );
+	}
+
+	private void storeData(Long id, Object entry) {
+		Work work = new Work( entry, id, WorkType.ADD, false );
+		TransactionContextForTest tc = new TransactionContextForTest();
+		sfHolder.getSearchFactory().getWorker().performWork( work, tc );
+		tc.end();
+	}
+
+	private void assertFacet(Facet facet, String expectedValue, int expectedCount) {
+		assertEquals( "Wrong facet value", expectedValue, facet.getValue() );
+		assertEquals( "Wrong facet count", expectedCount, facet.getCount() );
+	}
+
+	@Indexed
+	static class StringArrayFacetEntity {
```

Comment:
Any reason why they are not private? I wouldn't expect them to be used elsewhere in the package.